test/integration/bad/function-signatures/overloading/stanc.expected

```diff
@@ -0,0 +1,219 @@
+  $ ../../../../../../install/default/bin/stanc no_minimum_variadic.stan
+Semantic error in 'no_minimum_variadic.stan', line 82, column 37 to line 84, column 81:
+   -------------------------------------------------
+    80:                                                   gamma, xi, delta);
+    81:    // overloaded
+    82:    array[N_t] vector<lower=0>[4] y2 = ode_rk45_tol(simple_SIR, y0, t0, t, 1e-6,
+                                              ^
+    83:                                                  1e-6, 1000, beta, kappa,
+    84:                                                  gamma, xi, delta, unused, unused);
+   -------------------------------------------------
+
+No unique minimum promotion found for function "simple_SIR".
+Overloaded functions must not have multiple equally valid promotion paths.
+This function has several:
+(real, vector, real, real, real, real, real, real, int) => vector
+(real, vector, real, real, real, real, real, int, real) => vector
+Consider defining a new signature for the exact types needed or
+re-thinking existing definitions.
+  $ ../../../../../../install/default/bin/stanc overload_reduce_sum.stan
+Semantic error in 'overload_reduce_sum.stan', line 2, column 7 to column 17:
+   -------------------------------------------------
+     1:  functions {
+     2:    real reduce_sum(real x){
+                ^
+     3:      return x;
+     4:    }
+   -------------------------------------------------
+
+Identifier 'reduce_sum' clashes with a non-overloadable Stan Math library function.
+  $ ../../../../../../install/default/bin/stanc redefine-ccdf-same.stan
+Semantic error in 'redefine-ccdf-same.stan', line 2, column 2 to line 4, column 3:
+   -------------------------------------------------
+     1:  functions {
+     2:    real poisson_lccdf(int n, real x) {
+           ^
+     3:      return -x^2;
+     4:    }
+   -------------------------------------------------
+
+Function 'poisson_lccdf' has already been declared to for signature (int, real) => real
+  $ ../../../../../../install/default/bin/stanc redefine-ccdf-same2.stan
+Semantic error in 'redefine-ccdf-same2.stan', line 2, column 2 to line 4, column 3:
+   -------------------------------------------------
+     1:  functions {
+     2:    real poisson_lccdf(int n, real x) {
+           ^
+     3:      return -x^2;
+     4:    }
+   -------------------------------------------------
+
+Function 'poisson_lccdf' has already been declared to for signature (int, real) => real
+  $ ../../../../../../install/default/bin/stanc redefine-cdf-same.stan
+Semantic error in 'redefine-cdf-same.stan', line 2, column 2 to line 4, column 3:
+   -------------------------------------------------
+     1:  functions {
+     2:    real poisson_lcdf(int n, real x) {
+           ^
+     3:      return -x^2;
+     4:    }
+   -------------------------------------------------
+
+Function 'poisson_lcdf' has already been declared to for signature (int, real) => real
```

Comment:
Should we tell them it is defined in the math library?

Reply:
Yes. Easy and good for users